imap_processing/swe/l1b/swe_l1b_science.py

```diff
@@ -0,0 +1,509 @@
+    lookup_table_filepath = f"{imap_module_directory}/swe/l1b/swe_esa_lookup_table.csv"
+    lookup_table = pd.read_csv(
+        lookup_table_filepath,
+        index_col="e_step",
+    )
+
+    if table_index_value == 0:
+        return lookup_table.loc[lookup_table["table_index"] == 0]
+
+    if table_index_value == 1:
+        return lookup_table.loc[lookup_table["table_index"] == 1]
+
+    raise ValueError("Error: Invalid table index value")
+
+
+def deadtime_correction(counts: np.array, acq_duration: int):
+    """Calculate deadtime correction.
+
+    Deadtime correction is a technique used in various fields, including
+    nuclear physics,radiation detection, and particle counting, to compensate
+    for the effects of the time period during which a detector is not able to
+    record new events or measurements after detecting a previous event.
+    This "deadtime" is essentially the time during which the detector is
+    recovering from the previous detection and is unable to detect new events.
+
+    In particle detectors, there is a finite time required for the detector to
+    reset or recover after detecting a particle. During this deadtime, any
+    subsequent particles that may have arrived go undetected. As a result,
+    the recorded count rate appears to be lower than the actual count rate.
+
+    Deadtime correction involves mathematically adjusting the measured count
+    rates to compensate for this deadtime effect. This correction is crucial
+    when dealing with high-intensity sources or particle fluxes, as the deadtime
+    can significantly affect the accuracy of the measurements.
+
+    Deadtime correction is important to ensure accurate measurements and data
+    analysis in fields where event detection rates are high and where every
+    detected event is critical for understanding physical processes.
+
+    Parameters
+    ----------
+    counts : np.array
+        counts data before deadtime corrections
+    acq_duration : int
+        This is ACQ_DURATION from science packet
+
+    Returns
+    -------
+    np.array
+        Corrected counts
+    """
+    # deadtime will be constant once it's defined.
+    # This deadtime value is from previous mission. SWE
+    # will give new one once they have it ready.
+    # TODO: update deadtime when we get new number
+    deadtime = 1.5e-6
+    correct = 1.0 - (deadtime * counts / acq_duration)
+    correct = np.maximum(0.1, correct)
+    corrected_count = np.divide(counts, correct)
+    return corrected_count
+
+
+def convert_counts_to_rate(data: np.array, acq_duration: int):
+    """Convert counts to rate using sampling time.
+
+    acq_duration is ACQ_DURATION from science packet.
+
+
+    Parameters
+    ----------
+    data : np.array
+        counts data
+    acq_duration : int
+        Acquisition duration. acq_duration is in millieseconds
+
+    Returns
+    -------
+    np.array
+        Count rates array in seconds
+    """
+    return np.divide(data, acq_duration)
```

Comment:
no particular reason. I feel like I heard in the past that using np.divide is suggested.

Reply:
The only reason I'd possibly suggest avoiding it is because of some of the deferral mechanisms for other array libraries. i.e. if you do xarray.Dataset / np.ma.MaskedArray, using `np.divide()` can force those to numpy-arrays rather than letting the dataset and masked array choose whether to return a Dataset with metadata still attached instead. No concern with what you have here, but I would actually suggest the opposite in a more general sense.

Reply:
That's good to know. ty!